### PR TITLE
[SIG-2924] no error on upload of a small foto

### PR DESCRIPTION
**Pull request by aditudorache**
This PR solves the issue when no error was shown if a to small foto is uploaded in the incident form. For this, the error messages are moved to the FileInput state and a FileInputRenderer was added to enable using of the react hooks functionality

### CODE REVIEW

File: src/signals/incident/components/form/FileInput/index.js [hunk 1]
```diff
@@ -77,7 +94,7 @@ const FileInput = ({ handler, touched, hasError, getError, parent, meta, validat
         reader.readAsText(file);
       });
     }
-  };
+  }, []);
```
Comment (jpoppe) [inline src/signals/incident/components/form/FileInput/index.js:97]: Did you see this GitHub Lint warning?

**Comment by jpoppe** [comment]
As I said looks good, @janjaap could you please double check if I missed something?

File: src/signals/incident/components/form/FileInput/__tests__/FileInput.test.js [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+        expect(FileReader).toHaveBeenCalled();
+        expect(addEventListener).toHaveBeenCalledWith('load', expect.any(Function));
+        expect(readAsText).toHaveBeenCalled();
+        expect(parentControls['input-field-name'].updateValueAndValidity).toHaveBeenCalledTimes(2);
+        expect(parent.meta.updateIncident).toHaveBeenCalledWith({
+          'input-field-name_previews': expect.any(Array),
+          'input-field-name': [file4, file1],
+        });
+
+        jest.runAllTimers();
```
Comment (janjaap) [inline src/signals/incident/components/form/FileInput/__tests__/FileInput.test.js:209]: Which timers need to be run? I didn't see any set in the component.

File: src/signals/incident/components/form/FileInput/__tests__/FileInput.test.js [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+        expect(readAsText).toHaveBeenCalled();
+        expect(parentControls['input-field-name'].updateValueAndValidity).toHaveBeenCalledTimes(1);
+        expect(parent.meta.updateIncident).toHaveBeenCalledWith({
+          'input-field-name_previews': expect.any(Array),
+          'input-field-name': [file1],
+        });
+      });
+
+      it('it removes 1 file when remove button was clicked', async () => {
+        parent.value = {
```
Comment (janjaap) [inline src/signals/incident/components/form/FileInput/__tests__/FileInput.test.js:248]: I would strongly advise not to rewrite variables, but to destructure them as prop:
```jsx
<FileInput
  parent={{...parent, value: { ... } }}
  ...
/>
```

Comment (aditudorache) [reply src/signals/incident/components/form/FileInput/__tests__/FileInput.test.js:248]: Sure!

File: src/signals/incident/components/form/FileInput/index.js [hunk 4]
```diff
@@ -79,7 +93,7 @@ const FileInput = ({ handler, touched, hasError, getError, parent, meta, validat
...
```
[the remark this answers is not in the document]
Comment (aditudorache) [reply src/signals/incident/components/form/FileInput/index.js:96]: Strange, sinds the the new eslint rules, I get no warnings in the IDE

File: src/signals/incident/components/form/FileInput/index.js [hunk 5]
```diff
@@ -138,64 +151,70 @@ const FileInput = ({ handler, touched, hasError, getError, parent, meta, validat
   };
 
   const previews = (parent && parent.value && parent.value[`${meta && meta.name}_previews`]) || [];
-  const errors = (parent && parent.value && parent.value[`${meta && meta.name}_errors`]) || null;
   const numberOfEmtpy = maxNumberOfFiles - previews.length - 1;
   const empty = numberOfEmtpy < 0 ? [] : [...Array(numberOfEmtpy).keys()];
 
-  if (!meta?.isVisible) return null;
```
[the remark this answers is not in the document]
Comment (aditudorache) [reply src/signals/incident/components/form/FileInput/index.js:145]: Removed

File: src/signals/incident/components/form/FileInput/index.js [hunk 6]
```diff
@@ -138,64 +151,70 @@ const FileInput = ({ handler, touched, hasError, getError, parent, meta, validat
...
-
   return (
-    <Header meta={meta} options={validatorsOrOpts} touched={touched} hasError={hasError} getError={getError}>
-      <div className="file-input">
+    <Fragment>
+      <FileInputStyle className="file-input" data-testid="fileInput">
```
[the remark this answers is not in the document]
Comment (aditudorache) [reply src/signals/incident/components/form/FileInput/index.js:135]: Correct!

**Comment by janjaap** [comment]
@aditudorache LGTM. Tests fail, though.

File: src/signals/incident/components/form/FileInput/index.js [hunk 7]
```diff
@@ -56,6 +56,7 @@ const FileInput = ({ handler, parent, meta }) => {
 
   const handleChange = useCallback(
     event => {
+      /* istanbul ignore next */
       if (!event?.target?.files?.length) return;
```
Comment (janjaap) [inline src/signals/incident/components/form/FileInput/index.js:60]: Wouldn't this be an important part to test?

File: src/signals/incident/components/form/FileInput/index.js [hunk 8]
```diff
@@ -84,6 +85,7 @@ const FileInput = ({ handler, parent, meta }) => {
       });
 
       files.forEach((file, uploadBatchIndex) => {
+        /* istanbul ignore next */
         if (files[uploadBatchIndex].existing) {
```
Comment (janjaap) [inline src/signals/incident/components/form/FileInput/index.js:89]: Same here; isn't this an important part of the functionality to test?

**Pull request merged**